### PR TITLE
feat: [SSCA-261]: Add Orcestration API in ssca-manager

### PR DESCRIPTION
**Pull request by arpit-harness**
## Describe your changes
Add Orcestration API in ssca-manager. Added a new spec with tag "SbomProcessor" and provided its implementation.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)

### CODE REVIEW

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Review by deepharness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 8 files at r1, 1 of 3 files at r2, 1 of 4 files at r5, 8 of 26 files at r6, 4 of 10 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 14 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/SSCAManagerConfiguration.java` line 69 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2G3rQ-Dci2zL0sNPH:-NZ2G3rQ-Dci2zL0sNPI:bxcrfpp) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/SSCAManagerConfiguration.java#L69)):*
> ```Java
>     defaultSwaggerBundleConfiguration.setResourcePackage(RESOURCE_PACKAGE);
>     defaultSwaggerBundleConfiguration.setSchemes(new String[] {"https", "http"});
>     defaultSwaggerBundleConfiguration.setHost("{{localhost}}");
> ```

it should be hostname field
___
*[`ssca-manager/service/src/main/java/io/harness/repositories/NormalizedSBOMRepository.java` line 20 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2GK02A1tida71C3Ub:-NZ2GK02A1tida71C3Uc:b3017e9) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/repositories/NormalizedSBOMRepository.java#L20)):*
> ```Java
> @HarnessRepo
> @OwnedBy(SSCA)
> public interface NormalizedSBOMRepository
> ```

Can we call it SBOMComponentRepo ?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 55 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2PcKGBPSZHwKwQfWC:-NZ2PcKGBPSZHwKwQfWD:b-46ll5l) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L55)):*
> ```Java
>   public Response sbomProcess(
>       String orgIdentifier, String projectIdentifier, SbomProcessRequestBody sbomProcessRequestBody, String accountId) {
>     log.info(String.format("sbomRequestBody: %s", sbomProcessRequestBody));
> ```

You can look into https://github.com/sizovs/PipelinR its a simple CQRS framework which will simplify the logic which we write in controllers also make it more testable  
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 64 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2Kfqg5mdyRjpmXUPm:-NZ2Kfqg5mdyRjpmXUPn:b6rbhor) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L64)):*
> ```Java
>       log.error(String.format("Error in writing sbom to file: %s", sbomDumpFile));
>     }
>     // add upload to gcp step.
> ```

Is it TODO? 
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 70 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2NfyL65luksjuiotO:-NZ2NfyL65luksjuiotP:b-yo5i37) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L70)):*
> ```Java
>       SpdxDTO spdxDTO =
>           new Gson().fromJson(new String(sbomProcessRequestBody.getSbomProcess().getData()), SpdxDTO.class);
>       artifactEntity = artifactService.getArtifactFromSbomPayload(
> ```

Logic of checking the sbom type can be shifted to artifactService . It will make controller looks cleaner.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 92 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2GxlO9WFvk6eJK-N8:-NZ2GxlO9WFvk6eJK-N9:b-48lp3d) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L92)):*
> ```Java
>             .accountID(accountId)
>             .sbomURL(serverUrlConfigService.getApiBaseUrl() + "api/v1/artifacts/" + artifactEntity.getArtifactId()
>                 + "/stepExecutions/" + sbomProcessRequestBody.getSbomMetadata().getStepExecutionId() + "/sbom/"
> ```

Create a separate method for sbom url generation. 
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 108 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2HsJE-pC7j982w8PX:-NZ2HsJE-pC7j982w8PY:b-a604i1) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L108)):*
> ```Java
> 
>     for (NormalizedSBOMEntity sbomEntity : sbomEntityList) {
>       normalizedSBOMRepository.save(sbomEntity);
> ```

we should use bulkSave. saving each entity is too costlier for write.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/Normalizer.java` line 16 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2R24Y4rPJHte2erYq:-NZ2R24Y4rPJHte2erYr:bghnaod) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/Normalizer.java#L16)):*
> ```Java
> import java.util.List;
> 
> public interface Normalizer {
> ```

Should use generics Normalizer<SPDX/Cyclonedx>
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/Normalizer.java` line 17 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2RDDQCTYYhkWpPFcJ:-NZ2RDDQCTYYhkWpPFcK:brfrah3) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/Normalizer.java#L17)):*
> ```Java
> 
> public interface Normalizer {
>   List<NormalizedSBOMEntity> NormaliseSBOM(String DownloadLink, SettingsDTO settings) throws ParseException;
> ```

As we already have SPDX or cycloneDx dto we can change the signature to NormaliseSBOM(T dto, SettingsDTO settings)
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 82 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2LedT2pNdKu9Otsp5:-NZ2LedT2pNdKu9Otsp6:bygl7ye) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L82)):*
> ```Java
>         normalizedSBOMEntityBuilder.packageOriginatorName(spdxPackage.getOriginator());
>       }
>       CyclonedxNormalizer.getVersionInfo(spdxPackage.getVersionInfo(), normalizedSBOMEntityBuilder);
> ```

Why a SPDX normalizer is referring to CyclonedxNormalizer?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 87 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2SvzVD3yJWs_Dr8-M:-NZ2SvzVD3yJWs_Dr8-N:b1fl5mc) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L87)):*
> ```Java
>         getPackageManagerSpdx(spdxPackage, normalizedSBOMEntityBuilder);
>       } catch (InvalidArgumentsException e) {
>         continue;
> ```

do we have to log the exception ?
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 95 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2SLf7E4E5QVzV6H7m:-NZ2SLf7E4E5QVzV6H7n:bq888wh) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L95)):*
> ```Java
>   }
> 
>   private void getPackageManagerSpdx(
> ```

don't pass the reference to a method . Instead of passing the builder take the output parameter 
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 98 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2M2gM2k4jFa5UzmTt:-NZ2M2gM2k4jFa5UzmTu:b-x57bv1) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L98)):*
> ```Java
>       SpdxDTO.Package spdxPackage, NormalizedSBOMEntity.NormalizedSBOMEntityBuilder normalizedSBOMEntityBuilder) {
>     for (SpdxDTO.Package.ExternalRefs externalRef : spdxPackage.getExternalRefs()) {
>       if (externalRef.getReferenceCategory().equals("PACKAGE-MANAGER")) {
> ```

use constants "PACKAGE-MANAGER""
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 29 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2Jkuo0f2cv18YHslT:-NZ2Jkuo0f2cv18YHslU:bm9zuya) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L29)):*
> ```Java
>   public ArtifactEntity getArtifactFromSbomPayload(String accountId, String orgIdentifier, String projectIdentifier,
>       SbomProcessRequestBody body, String sbomVersion) {
>     String artifactId = UUID.randomUUID().toString();
> ```

artifactId should not be random , it should be generated from RegistryUrl
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 58 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2J7NG5yNU6_GprigZ:-NZ2J7NG5yNU6_Gprig_:b3pg0m) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L58)):*
> ```Java
>                   .sbomUrl(serverUrlConfigService.getApiBaseUrl() + "api/v1/artifacts/" + artifactId
>                       + "/stepExecutions/" + body.getSbomMetadata().getStepExecutionId() + "/sbom/"
>                       + body.getSbomProcess().getName() + "?accountIdentifier=" + accountId
> ```

sbom url logic is duplicated take it from a utils

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 17 file(s) to review. 1 change requests. 11. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/SSCAManagerConfiguration.java` line 69 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2G3rQ-Dci2zL0sNPH:-NZ4W-_x7549aobgtqac:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/SSCAManagerConfiguration.java#L69)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

it should be hostname field
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 55 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2PcKGBPSZHwKwQfWC:-NZ4XQZC2XpgGio7ugwk:b-7wl8ep) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L55)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

You can look into https://github.com/sizovs/PipelinR its a simple CQRS framework which will simplify the logic which we write in controllers also make it more testable  
</blockquote></details>

checking.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 64 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2Kfqg5mdyRjpmXUPm:-NZ4XSi8BIbWu2pvOaLe:b-3ssen1) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L64)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Is it TODO? 
</blockquote></details>

Yes is a TODO, Will mark it so. Will do it now only with review comments.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 70 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2NfyL65luksjuiotO:-NZ4dmpR1J74zSGr9Pdb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L70)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Logic of checking the sbom type can be shifted to artifactService . It will make controller looks cleaner.
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 92 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2GxlO9WFvk6eJK-N8:-NZ4eJs45fIqPxB_LcJm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L92)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Create a separate method for sbom url generation. 
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 82 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2LedT2pNdKu9Otsp5:-NZ58Kp14UEv8LkqWLik:b1xaill) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L82)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Why a SPDX normalizer is referring to CyclonedxNormalizer?
</blockquote></details>

Done. Moved to SBOMUtils.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 87 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2SvzVD3yJWs_Dr8-M:-NZ4fWXN8utjb6duyk9_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L87)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

do we have to log the exception ?
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 98 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2M2gM2k4jFa5UzmTt:-NZ5G3gc1I3MDhrGwrms:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L98)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

use constants "PACKAGE-MANAGER""
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 29 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2Jkuo0f2cv18YHslT:-NZ4sNvdEBx4BdhDvjmH:b55l66o) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L29)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

artifactId should not be random , it should be generated from RegistryUrl
</blockquote></details>

Its Only random in case any artifact with that registryUrl doesn't exist.. else we take the already existing UUID.
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java` line 58 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2J7NG5yNU6_GprigZ:-NZ57tSf49EGE-xYn1KK:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/services/ArtifactServiceImpl.java#L58)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

sbom url logic is duplicated take it from a utils 
</blockquote></details>

Done.
___
*[`ssca-manager/service/src/main/java/io/harness/repositories/NormalizedSBOMRepository.java` line 20 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2GK02A1tida71C3Ub:-NZ4W1LfBhFf52Bl9g2v:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/repositories/NormalizedSBOMRepository.java#L20)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Can we call it SBOMComponentRepo ?
</blockquote></details>

Done.

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 17 file(s) to review. 1 change requests. 12. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java` line 108 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2HsJE-pC7j982w8PX:-NZ5Yat45TdY0G-wdZjb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/api/SbomProcessorApiImpl.java#L108)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

we should use bulkSave. saving each entity is too costlier for write.
</blockquote></details>

Done.

File: ssca-manager/service/src/main/java/io/harness/ssca/normalize/CyclonedxNormalizer.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.util.Strings;
+
+@Slf4j
+public class CyclonedxNormalizer implements Normalizer {
+  @Override
+  public List<NormalizedSBOMComponentEntity> NormaliseSBOM(String downloadLink, SettingsDTO settings)
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/ssca/normalize/CyclonedxNormalizer.java:31]: method name should be small

**Comment by arpit-harness** [comment]
trigger uts

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+          type: string
+          description: id of the step
+          example: log4Jj
+        build_url:
+          type: string
+          description: BuildURL
+          example: log4Jj
+        step_execution_id:
+          type: string
+          description: StepExecutionId
```
Comment (udayvunnam) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:158]: you can give valid sample details here

**Review by deepharness — CHANGES_REQUESTED** [review]
> Dismissed @deepharness from a discussion.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 24 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java` line 95 at r7](https://reviewable.io/reviews/harness/harness-core/49361#-NZ2SLf7E4E5QVzV6H7m:-NZ9gCDU2qAcCe5TnmvG:b-guu51q) ([raw file](https://github.com/harness/harness-core/blob/c5f4f28de373be3954dd427d7693f56401586c88/ssca-manager/service/src/main/java/io/harness/ssca/normalize/SpdxNormalizer.java#L95)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

don't pass the reference to a method . Instead of passing the builder take the output parameter 
</blockquote></details>

I mean signature should be like  string getFormattedPurl SpdxDTO.Package spdxPackage) 
 then 

builder.Purl(getgetFormattedPurl(spdxDto);

File: ssca-manager/contracts/openapi/v1/openapi.yaml (hunk 2 above)
Comment (arpit-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:158]: Done.

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger ut0

**Comment by arpit-harness** [comment]
trigger ut6

**Comment by arpit-harness** [comment]
trigger ut0

**Comment by arpit-harness** [comment]
trigger ut0

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

**Review by deepharness — APPROVED** [review]
> Reviewed 1 of 26 files at r6, 2 of 19 files at r10, 1 of 10 files at r14.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 24 file(s) to review. 1. unresolved discussions.

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+servers:
+  - url: 'https://app.harness.io'
+    description: Harness host URL.
+paths:
+  '/v1/org/{org}/project/{project}/sbom/process':
+    parameters:
+      - $ref: '#/components/parameters/OrgParam'
+      - $ref: '#/components/parameters/ProjectParam'
+    post:
+      summary: ''
```
Comment (mankrit-singh) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:19]: Please add a summary here. In Stoplight it will be under "Operation Name"

**Review by mankrit-singh — APPROVED** [review]

**Comment by arpit-harness** [comment]
trigger uts

**Review by deepharness — APPROVED** [review]
> Reviewed 1 of 26 files at r6, 1 of 10 files at r7, 2 of 3 files at r9, 7 of 19 files at r10, 3 of 7 files at r11, 9 of 10 files at r14, 1 of 1 files at r15, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49361)* status: 1. unresolved discussions.

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger ut6

**Pull request merged**